bridge_adaptivity/module/static/module/js/module.js

```diff
@@ -1,20 +1,183 @@
+                    localStorage.setItem("bridgeState", JSON.stringify(this));
+                    console.debug("State saved: ", this);
+                }
+            };
+        }());
+
+        bridgeState.load();
+
+        // activity source preview:
+        $.each(activitiesData, function (i, activity) {
+            createPreviewButton(
+                activity["name"],
+                activity["launch_url"],
+                activity["id"],
+                $("#activity-row-" + i + " td").last(),
+                modalContentFrame
+            )
+        });
+
+        $("#accordion").find("a[data-course-id]").on("click", function () {
+            var target = $(this);
+            var activeCourseIndex = target.data("course-index");
+            var activeCourseId = target.data("course-id");
+
+            if (activeCourseIndex !== bridgeState.accordion.opened) {
+                bridgeState.accordion.opened = activeCourseIndex;
+                bridgeState.save();
+                clickHandler(activeCourseIndex, activeCourseId, bridgeState);
+            } else {
+                bridgeState.accordion.opened = null;
+                bridgeState.save();
+            }
+        });
 
-    // after timeout, follow the next activity link
-    function afterTimeout(){
-        console.log(next_button.attr('href'));
-        window.location = next_button.attr('href');
-    }
+        function clickHandler(activeCourseIndex, activeCourseId, state) {
+            var courseData;
+            var content_panel = $("#content-panel-" + activeCourseIndex + " div.panel-body");
+            if (state.accordion.courseData[activeCourseIndex] !== undefined) {
+                console.log("Taking data from cache ...");
+                courseData= state.accordion.courseData[activeCourseIndex];
+                renderCourseBlocks(courseData, content_panel);
+            } else {
+                var requestData = {course_id: activeCourseId};
+                $.post(internalUrls.apiSources, requestData, function (responseData) {
+                    console.log("Processing API request...");
+                    state.accordion.courseData[activeCourseIndex] = responseData;
+                    state.accordion.activeCourseId = activeCourseId;
+                    state.save();
+                    renderCourseBlocks(responseData, content_panel);
+                });
+            }
+        }
 
-});
+        function renderCourseBlocks(courseData, container) {
+            console.log("Rendering course...");
+            var usedLtiUrls = $.map(activitiesData, function(data) { return data.launch_url });
+            var sourcesList = $("<div/>").addClass("list-group");
+            $.each(courseData, function (i, item) {
+                var listItem = $("<button/>")
+                    .addClass("list-group-item")
+                    .attr("type", "button")
+                    .appendTo(sourcesList);
+                $("<span/>")
```

Comment:
It would be great if we append icon's span to the sourceButton action so click on icon also adds activity to the collection